pkg/apis/serving/v1alpha1/kfservice_types.go

```diff
@@ -42,6 +41,29 @@ type ModelSpec struct {
 	XGBoost    *XGBoostSpec    `json:"xgboost,omitempty"`
 	SKLearn    *SKLearnSpec    `json:"sklearn,omitempty"`
 	PyTorch    *PyTorchSpec    `json:"pytorch,omitempty"`
+	Explain    *ExplainerSpec  `json:"explain,omitempty"`
+}
+
+// ExplainerSpec defines the arguments for a model explanation server
+type ExplainerSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one spec.
+	Alibi  *AlibiExplainerSpec `json:"alibi,omitempty"`
+	Custom *CustomSpec         `json:"custom,omitempty"`
+}
+
+// AlibiExplainerSpec defines the arguments for configuring an Alibi explanation serving
+type AlibiExplainerSpec struct {
+	// The following fields follow a "1-of" semantic. Users must specify exactly one of Type or ExplainerURI.
+	// The kind of Alibi explanation server to create, e.g. AnchorTabular
+	Type string `json:"type"`
+	// The location of a fit explanation model
+	SavedExplainerURI string `json:"savedExplainerUri,omitempty"`
+	// Defaults to latest Alibi Version.
+	RuntimeVersion string `json:"runtimeVersion,omitempty"`
+	// Defaults to requests and limits of 1CPU, 2Gb MEM.
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+	// Inline custom parameter settings for explainer
+	Config v1.ConfigMap `json:"config,omitempty"`
```

Comment:
I'm pretty sure configmap is the wrong concept here. Can you provide another example of someone passing a configmap through an interface like this? I think the better abstraction is probably a map<string,string>. 

Can you provide an example of what types of parameters would be passed through here?

Reply:
What inputs we are expecting in Map? Also this sentence _'AlibiExplainerSpec defines the arguments for configuring an Alibi explanation serving'_ - not sure if its coming across in a cohesive way?

With this _'Alibi explanation server'_ -  I believe then the agreement yesterday was that explainer is another server in KFServing spec? Tomorrow if we have 'Bias Detector', I presume that will be another server on same lines?

Reply:
@ellis-bigelow Some example parameters for AnchorTabular can be seen [here](https://docs.seldon.io/projects/alibi/en/v0.2.2/api/alibi.explainers.anchor_tabular.html#alibi.explainers.anchor_tabular.AnchorTabular.explain). 

A Map is also probably fine.

Reply:
@animeshsingh We are handling explainers in a synchronous way only so any other component that can be used in this way could also be added eventually (see comment on levels below).

Reply:
I think you can replace Config v1.ConfigMap as `Parameters []ParameterSpec`

type ParameterSpec struct {
  key String
  value String
}

Reply:
@cliveseldon, should we just wire all those parameters from  https://docs.seldon.io/projects/alibi/en/v0.2.2/api/alibi.explainers.anchor_tabular.html#alibi.explainers.anchor_tabular.AnchorTabular.explain? We could even allow a map for kwargs.

Reply:
I think for now we can stick with a map. Once things settle we could consider maybe the most important explicitly. The issue is there could be a large number which may also change more frequently that the rest of the Spec here.